engine/src/main/java/com/sandboni/core/engine/finder/JarFinder.java

```diff
@@ -19,43 +20,45 @@
 import java.util.*;
 import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
+import java.util.stream.Stream;
 
 public class JarFinder extends FileTreeFinder {
     private static final Logger log = LoggerFactory.getLogger(JarFinder.class);
-    private Collection<ClassVisitor> visitors;
-
-    public JarFinder(ClassVisitor[] visitors) {
-        this.visitors = Collections.unmodifiableCollection(Arrays.asList(visitors));
-    }
 
     @Override
     protected Map<String, ThrowingBiConsumer<File, Context>> getConsumers() {
         ThrowingFunction<ClassParser, JavaClass> parse = ClassParser::parse;
         HashMap<String, ThrowingBiConsumer<File, Context>> map = new HashMap<>();
         map.put(ExtensionType.JAR.type(), (file, context) -> {
+            log.info("[{}] Jar scanning starts for {}", Thread.currentThread().getName(), file);
+            long start = System.nanoTime();
             try (JarFile jar = new JarFile(file)) {
                 Collections.list(jar.entries()).stream()
                         .filter(e -> !e.isDirectory())
                         .forEach(e -> {
                             if (e.getName().endsWith(ExtensionType.FEATURE.type())) {
                                 List<Link> links = CucumberFeatureFinder.toLinks(context, new File(file.getAbsolutePath() + File.separator + e.getName()), getFileContent(jar, e));
                                 context.addLinks(links.toArray(new Link[0]));
-                            } else if (e.getName().endsWith(ExtensionType.CLASS.type())) {
+                            } else if (e.getName().endsWith(ExtensionType.CLASS.type()) && context.inScope(e.getName())) {
                                 ClassParser classParser = new ClassParser(file.getAbsolutePath(), e.getName());
                                 JavaClass jc = parse.apply(classParser);
-                                if (context.inScope(jc.getClassName())) {
-                                    Link[] links = startVisitors(jc, context);
-                                    context.addLinks(links);
-                                }
+                                Link[] links = startVisitors(jc, context);
+                                context.addLinks(links);
                             }
                         });
             }
+            log.info("[{}] Jar scanning finished for {} in {} milliseconds", Thread.currentThread().getName(), file, TimeUtils.elapsedTime(start));
         });
         return map;
     }
 
+    @Override
+    protected boolean scanDependencies() {
+        return true;
+    }
+
     private Link[] startVisitors(JavaClass jc, Context c) {
-        return visitors.stream().flatMap(v -> v.start(jc, c)).toArray(Link[]::new);
+        return Stream.of(ProcessorBuilder.getVisitors()).flatMap(v -> v.start(jc, c)).toArray(Link[]::new);
```

Comment:
I think `Stream.of()` calls to `Arrays.stream()` for non primitives, maybe worth using the concrete `Arrays.stream()`?

Reply:
I don't see a difference in the usage because this is not an array of primitives, seems more like a code style approach. I'll change it.